### DISCUSSION TITLE
How to deploy the solution

### DISCUSSION THREAD

**Discussion by MasterBehind** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#57] [Q&A]
Hi,
I have created meta developer accout, app, and added whatsapp product with the right permissions, created user permenant token.
I got stuck into the following two points

- Configuring Web Hook
- In relation to the above, i need to setup the callback URL, which means, correct me if i am wrong, deploy this sample solution in my site, for this i dont have the right experience to do so, so i need some one to help me in step-by-step on how to do that.

much appreciated in advance

**Comment by gabrieldwight** [comment]
Hi, 

if you are using azure cloud service you can publish directly from visual studio using web deploy publishing settings which you can get from creating an app service. Here is the link for more information [Deploy ASP.NET Core apps to Azure App Service](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/azure-apps/?view=aspnetcore-8.0&tabs=visual-studio)

if you are using other hosting services, you have to check if you can host asp net core apps on their platform and also ensure .net runtime is installed there.

You can also host the solution with azure functions if possible.

The callback URL will work as long as your solution is hosted and secure with ssl.

**Comment by MasterBehind** [comment]
Hi,

I am using normal web hosting, they have asp.net, and i am running another .net application overthere, so it has .net runtime as well, now i have created one virtual direcotry https://www.print-ley.com/EndPoint, but it hroughs error as below
<img width="786" alt="image" src="https://github.com/gabrieldwight/Whatsapp-Business-Cloud-Api-Net/assets/152811181/8218102a-8411-4986-bcc5-7e13a2ef8c7b">
Any thing i need to do

**Reply by gabrieldwight** [reply]
Mostly it can be caused when the web hosting does not have the latest .net runtime installed. You can request them to install for you the latest runtime or you can publish with the runtime using self contained option

**Comment by MasterBehind** [comment]
tried with self-contained but still getting the same error, BTW, what should i put as "Verify Token", and also what are the other possiblities of such issue?

**Reply by gabrieldwight** [reply]
For the startup error, you can check with the hosting company to solve the issue of the .net runtime. 

In the verify token you can put any unique string that you will send from meta as long as it matches the string specified in the endpoint it will subscribe to your webhook URL for receiving messages. It is only done once until you decide to change the endpoint to repeat the same process.

**Reply by MasterBehind** [reply]
I fixed it now, however, i got new error message
<img width="935" alt="image" src="https://github.com/gabrieldwight/Whatsapp-Business-Cloud-Api-Net/assets/152811181/88002b6c-257f-4281-919a-338999d5a4df">

**Reply by gabrieldwight** [reply]
You can check if IIS is configured to access the directory that contain wwwroot

**Reply by MasterBehind** [reply]
that issue fixed, but now we have another error
The URL couldn't be validated. Response does not match challenge, expected value="763248515", received="\u003C!DOCTYPE html>\n\u003Ch..."

Also, another question, in which part of the solution can I put the configuration values of the token, secret value, etc

**Reply by gabrieldwight** [reply]
For url validation check the token string you pass from meta matches with the string you specified in the code.

You can put within code if your code is not shared to untrusted sources or use azure key vault

**Reply by MasterBehind** [reply]
Where in the code? Which file?

**Reply by MasterBehind** [reply]
I did not specify any string in the code, i took your code as is and published it

**Reply by gabrieldwight** [reply]
My code sample was just for testing but you cn check on the web hook controller

**Reply by gabrieldwight** [reply]
You will see a get request method that contains parameter related to hub token string

**Reply by gabrieldwight** [reply]
You will see a get request method that contains parameter related to hub token string

**Comment by MasterBehind** [comment]
Hi Gabriel
Now the solution is working fine as per the image below, but i recieved nothing on my mobile

<img width="319" alt="image" src="https://github.com/gabrieldwight/Whatsapp-Business-Cloud-Api-Net/assets/152811181/158653c7-c18f-4c96-a270-834c4b4bf156">

Regards

**Reply by gabrieldwight** [reply]
if you are testing with test phone number, you will need to send the hello world template message from meta developer portal

**Comment by MasterBehind** [comment]
please note that my call back URL is 
https://print-ley.com/whatsapp

and i am trying to setup a webhook on the above URL but it shows this error 
"The URL couldn't be validated. Response does not match challenge, expected value="481395359", received="\u003C!DOCTYPE html>\n\u003Ch..."

**Comment by MasterBehind** [comment]
Where is the problem?

**Reply by gabrieldwight** [reply]
It needs that value; 481395359 passed as the verify token
